Propagator cleanup and minor fixes

hightower8083 opened the pull request:
updated copy of  #85

hightower8083 committed:
have merged move_time_window() method into propagate()
added `azimuthal_modes` as global box attribute and removed the duplicates
restyle of the propagator sequence to avoid multiple `if .. else`
added per-operator comments to the propagator methods
fixed backend to `numpy` only
suppressed the backend initiation messages from propagator construction

MaxThevenet commented on lasy/laser.py:
```diff
@@ -152,44 +128,59 @@
+
+        # Translate the phase of spectral image
+        self.field.field_fft *= np.exp(-1j * translate_time
+                                * self.field.omega.reshape(omega_shape))
```
Could you check the CodeQL comment?

hightower8083 replied:
yeah, fixed this rather theoretical risk =) 
it's also a general question -- if we hard-fix available geoms list to `xyt` and `rt` or keep the it open, and thus use open if-elif forks for `self.dim` with NonImplemented errors

MaxThevenet commented on lasy/laser.py:
```diff
@@ -152,44 +128,59 @@ def propagate(self, distance, nr_boundary=16):
+            for i_m in range(self.box.azimuthal_modes.size):
+                self.field.field_fft[i_m] = self.prop[i_m].step(
+                        self.field.field_fft[i_m], distance, overwrite=True)
         elif self.dim == 'xyt':
```
Should this be an `else`? Or should there be one at the end?
```suggestion
        else:
```

hightower8083 committed:
chaned open if-elif forks to if-else to fix the CodeQL test crash

hightower8083 committed:
moved time<>frequency transforms inside the propagator
updated comments
added backend as an optional argument
extended boundary absorbion option to `txy` geometry and set it to default None
updated get_field function to check different slices

hightower8083 committed:
Merge branch 'development' into propagator-cleanup

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci
